challenge-01/README.md

```diff
@@ -2,41 +2,47 @@
 
 ```js
 // Declarar uma variável chamada `myvar`, sem valor.
-?
+var myvar;
 
 // Após declarada, atribua o valor 10 à variável `myvar`.
-?
+myvar = 10;
 
 // Declare uma nova variável chamada `soma`, e adicione uma instrução somando os valores 15 e 8.
-?
+soma = 15 + 8;
 
 // Atribua à variável `soma` todo o valor dela, somando 1, usando o operador de soma abreviado.
-?
+soma = soma++;
```

Comment:
Sim, entendi perfeitamente. Falta de atenção mesmo na hora de desenvolver o desafio.
= D


Date: Mon, 30 Mar 2015 19:13:38 -0700
From: notifications@github.com
To: curso-javascript-ninja@noreply.github.com
CC: allexcarvalho@outlook.com
Subject: Re: [curso-javascript-ninja] Update README.md (#4)

In challenge-01/README.md:

>  
>  // Atribua à variável `soma` todo o valor dela, somando 1, usando o operador de soma abreviado.
> -?
> +soma = soma++;

@allexcarvalho, quando você usa esse formato soma = soma++, o resultado não é exatamente o que esperamos. Vou explicar: 


O operdor de soma abreviado ++ já causa um efeito colateral, pois ele já atribui à variável tudo que ela tem + 1.

Se você usa o operador de pré-incremento (++soma), o valor é atribuído à variável soma antes da operação. Se você usa o pós-incremento(soma++), primeiro a operação é executada (chamada da variável soma nesse caso), e então o valor de soma + 1 é atribuído à variável soma.


Exemplo:


var soma = 1;
var soma2 = soma++;

Após essa operação, soma2 vai receber o valor de soma, que é 1, enquanto a variável soma será incrementada com tudo que ela tem + 1.


Logo:


soma2; // => 1
soma; // => 2

Se eu usar o pré-incremento:


var soma = 1;
var soma2 = ++soma;

Nesse caso, primeiro o valor de soma + 1 é atribuído à soma, e então, o valor de soma (já com o novo valor), é atribuído à soma2.


O resultado é o seguinte:


soma; // => 2
soma2; // => 2

Compreendeu a diferença?


Só usando o soma++, ele já atribui e incrementa 1 no valor de soma :)


—
Reply to this email directly or view it on GitHub.